xdcs-server/src/main/java/pl/edu/agh/xdcs/services/sweeper/TaskSweeper.java

```diff
@@ -116,16 +130,87 @@ private void startTask(QueuedTaskEntity task, RuntimeTaskEntity runtimeTask, Lis
                                     .setAgentCount(agents.size())
                                     .setAgentId(agentId)
                                     .build()
-                    )
-                    .build());
+                    );
+            if (shouldPrepareKernelConfig(task.getHistoricalTask())) {
+                taskSubmit.setKernelConfig(prepareKernelConfig(task.getHistoricalTask(), kernelArguments));
+            }
+            taskRunner.submit(taskSubmit.build());
             agentId++;
         }
     }
 
+    private KernelConfig prepareKernelConfig(HistoricalTaskEntity task, List<KernelArgument> kernelArguments) {
+        KernelConfig.Builder kernelConfig = KernelConfig.newBuilder()
+                .setGlobalWorkShape(task.getGlobalWorkShape())
+                .addAllKernelArguments(kernelArguments);
+        if (task.getLocalWorkShape() != null) {
+            kernelConfig.setLocalWorkShape(task.getLocalWorkShape());
+        }
+        return kernelConfig.build();
+    }
+
     private Collection<String> getAgentIps(Set<AgentEntity> agents) {
         return agents.stream()
                 .map(AgentEntity::getAddress)
                 .map(InetAddress::getHostAddress)
                 .collect(Collectors.toList());
     }
+
+    private boolean shouldPrepareKernelConfig(HistoricalTaskEntity task) {
+        return EnumSet.of(TaskType.CUDA, TaskType.OPENCL).contains(
+                task.getDeploymentDescriptor().getDefinition().getType()
+        );
+    }
+
+    private List<KernelArgument> prepareKernelArguments(HistoricalTaskEntity task) {
+        int position = 0;
+        List<KernelArgument> result = Lists.newLinkedList();
+        for (KernelParameter kernelParameter : task.getDeploymentDescriptor().getDefinition().getKernelParams().getParameters()) {
+            result.add(
+                    KernelArgument.newBuilder()
+                            .setName(kernelParameter.getName())
+                            .setType(toGrpcEntity(kernelParameter.getType()))
+                            .setDirection(toGrpcEntity(kernelParameter.getDirection()))
+                            .setPosition(position)
+                            .setValue(getValueAtPosition(task, position))
+                            .build()
+            );
+            position++;
+        }
+        return result;
+    }
+
+    private String getValueAtPosition(HistoricalTaskEntity task, int position) {
+        for (KernelArgumentEntity kernelArgument : task.getKernelArguments()) {
+            if (kernelArgument.getPosition() == position) {
+                return kernelArgument.getValue().getReferencedObjectId();
+            }
+        }
+        throw new NoSuchElementException("There is no argument with position " + position +
+                " in task with id: " + task.getId());
+    }
+
+    private KernelArgument.Type toGrpcEntity(KernelParameter.Type model) {
```

Comment:
to można wynieść do osobnych mapperów

Reply:
ten kod został usunięty podczas zamiany w grpc z repeated KernelArgument na repeated string